lua/obsidian/picker_utils.lua

```diff
@@ -16,12 +16,39 @@ M.telescope_mappings = function(map, client, initial_query)
       require("telescope.actions").close(prompt_bufnr)
       client:command("ObsidianNew", { args = query })
     end,
+
+    obsidian_insert_link = function(prompt_bufnr)
+      local selected_path = require("telescope.actions.state").get_selected_entry().path
+      local rel_path = assert(client:vault_relative_path(selected_path))
+      if vim.endswith(rel_path, ".md") then
+        rel_path = string.sub(rel_path, 1, -4)
+      end
+      require("telescope.actions").close(prompt_bufnr)
+      local link
+      local note_id = require('obsidian').Note.from_file(rel_path .. ".md", client.dir).id
+      if client.opts.completion.preferred_link_style == "markdown" then
+        link = "["..note_id.."](" .. rel_path .. ".md)"
+      else
+        if client.opts.completion.prepend_note_id then
+          link = "[[" .. note_id .. "]]"
+        else
+          link = "[[" .. rel_path .. "]]"
+        end
+      end
```

Comment:
In 27bc40c I added a help function to format links, so you can replace this with:

```suggestion
      local note = require("obsidian").Note.from_file(path, client.dir)
      local link = client:format_link(note)
```

Reply:
A good idea to provide a format_link function. I changed my commit so it makes use of this function (and also the new config enum obsidian.config.LinkStyle.